operator/src/main/resources/scripts/stop-server.py

```diff
@@ -47,42 +205,60 @@ def getEnvVar(var):
 file.write(decoded)
 file.close()
 
-def shutdownUsingNodeManager(domainName, domainDir):
-  try:
-    nmConnect(userConfigFile='/weblogic-operator/introspector/userConfigNodeManager.secure',
-              host='127.0.0.1',port='5556',
-              domainName=domainName,
-              domainDir=domainDir,
-              nmType='plain')
-  except Exception, e:
-    print e
-    print('Failed to connect to the node manager')
-    exit(exitcode=2)
+# check if Coherence cluster exists in this domain
+cohExists = doesCoherenceExist()
 
-# Connect to the server and request that it shuts down
+# If Coherence exists then we need to connect to admin server, else local server
+if (cohExists):
+  print ('Shutdown: Coherence cluster exists')
+  connect_url = local_admin_protocol + '://' + admin_host + ':' + admin_port
 
-try:
-  connect(userConfigFile='/weblogic-operator/introspector/userConfigNodeManager.secure',
+  # must use force shutdown for admin server since Coherence MBeans cannot be found after
+  # a graceful admin server restart
```

Comment:
Is there a bug tracking this? Do you think all customers will be fine with this behavior? Maybe there are customers that'd prefer to have a graceful shutdown here when Cohernece is the picture, and pay the consequent penalty of having managed server shutdown extend to the full timeout...

Reply:
This is a Mac only issue, not seen on OKE.  It has to do with RMI timeout